src/main/java/eu/europeana/apikey/mail/MailService.java

```diff
@@ -20,28 +23,84 @@
+            message.setBcc(Objects.requireNonNull(template.getBcc()));
+            message.setTo(Objects.requireNonNull(template.getTo())[0]);
+            message.setSubject(Objects.requireNonNull(template.getSubject()));
             message.setText(messageBody);
 
             emailSender.send(message);
+        } catch (NullPointerException npe) {
+            throw new SendMailException("Missing required parameters prevented sending confirmation email", npe);
         } catch (MailException e) {
-            throw new SendMailException(
-                    String.format("A problem prevented sending a confirmation '%s' email to %s", subject, to), e);
+            throw new SendMailException(String.format("A problem prevented sending a confirmation '%s' email to %s",
+                                                      template.getSubject(),
+                                                      Objects.requireNonNull(template.getTo())[0]), e);
+        }
+    }
+
+    public void sendApiKeyEmail(SimpleMailMessage template,
+                                String firstName,
+                                String lastName,
+                                String apiKey) throws SendMailException {
+        String messageBody = String.format(Objects.requireNonNull(template.getText()), firstName, lastName, apiKey);
+        sendSimpleMessage(template, messageBody);
+    }
+
+    public void sendApiKeyAndClientEmail(SimpleMailMessage template,
+                                         String firstName,
+                                         String lastName,
+                                         String apiKey,
+                                         String clientSecret) throws SendMailException {
+        String messageBody = String.format(Objects.requireNonNull(template.getText()),
+                                           firstName,
+                                           lastName,
+                                           apiKey,
+                                           clientSecret);
+        sendSimpleMessage(template, messageBody);
+    }
+
+    public boolean sendDeletedUserEmail(SimpleMailMessage template,
+                                        String today,
+                                        String email,
+                                        String kcDeleted,
+                                        String setsDeleted,
+                                        String inThirtyDays) {
+        String messageBody = String.format(Objects.requireNonNull(template.getText()),
+                                           today,
+                                           email,
+                                           kcDeleted,
+                                           setsDeleted,
+                                           inThirtyDays);
+        try {
+            sendSimpleMessage(template, messageBody);
+        } catch (SendMailException sme) {
+            return false;
```

Comment:
I propose to log the exception to we can find out quickly what the root cause is when this happens